File: python/bob/bob.py
import re


def response(hey_bob: str):
    no_whitespaces = re.sub(r'\s', '', hey_bob)
    if len(no_whitespaces) == 0:
        print(len(no_whitespaces))
        return 'Fine. Be that way!'
    question = no_whitespaces.endswith('?')
    chars_only = re.sub(r'[\W\d]', '', hey_bob)
    shouting = re.fullmatch(r'[A-Z]+', chars_only)
    if question and shouting:
        return 'Calm down, I know what I\'m doing!'
    if question:
        return 'Sure.'
    if shouting:
        return "Whoa, chill out!"
    return "Whatever."

File: typescript/bob/bob.ts
export function hey(msg: string): string {
  // Strip whitespaces
  msg = msg.replaceAll(/\s/g, "");

  if (msg.length === 0) return "Fine. Be that way!";

  const question = msg.endsWith("?");

  // Strip non-letter characters
  msg = msg.replaceAll(/[\W\d]/g, "");

  const allCaps = /^[A-Z]+$/.test(msg);

  if (question && allCaps) return "Calm down, I know what I'm doing!";
  if (question) return "Sure.";
  if (allCaps) return "Whoa, chill out!";
  return "Whatever.";
}
